src/libstd/sys/windows/fs.rs

```diff
@@ -338,6 +338,12 @@ impl File {
         Ok(newpos as u64)
     }
 
+    pub fn duplicate(&self) -> io::Result<File> {
+        Ok(File {
+            handle: try!(self.handle.duplicate(0, true, c::DUPLICATE_SAME_ACCESS)),
```

Comment:
Asking 3.5 years after the fact: Does anyone know why we set `inheritable = true` here? I've run into a bug (because [`duct`](https://github.com/oconnor663/duct.rs) was abusing `File::try_clone` somewhat) where pipes get inherited by processes that aren't supposed to inherit them, when spawning happens on multiple threads at once. Normally the `CREATE_PROCESS_LOCK` prevents this, but this line here creates inheritable handles outside of that lock. The fix for `duct` is going to be to stop abusing `try_clone` and to just make all the appropriate system calls explicitly, which I should've done anyway. But this inheritable flag might be cause some programs in the wild to keep files open longer than they should (even without a multi-threaded race condition like mine), and on Windows in particular that could cause bugs by making them un-deletable.